scipy/signal/_arraytools.py

```diff
@@ -232,10 +228,11 @@ def zero_ext(x, n, axis=-1):
     array([[ 0,  0,  1,  2,  3,  4,  5,  0,  0],
            [ 0,  0,  0,  1,  4,  9, 16,  0,  0]])
     """
+    xp = _get_namespace(x)
     if n < 1:
         return x
     zeros_shape = list(x.shape)
     zeros_shape[axis] = n
-    zeros = np.zeros(zeros_shape, dtype=x.dtype)
-    ext = np.concatenate((zeros, x, zeros), axis=axis)
+    zeros = xp.zeros(zeros_shape, dtype=x.dtype)
```

Comment:
This is supposed to fail, right, if `xp` doesn't have `zeros` attribute/method/function?

Reply:
If a library is Array API compliant then it must have the `zeros` function, https://data-apis.org/array-api/latest/API_specification/generated/signatures.creation_functions.zeros.html